python/BioSimSpace/Sandpit/Exscientia/_SireWrappers/_molecule.py

```diff
@@ -1366,6 +1376,22 @@ def repartitionHydrogenMass(self, factor=4, water="no",
             self._sire_object = _SireIO.repartitionHydrogenMass(
                     self._sire_object, factor, water_options[water], pmap)
 
+            # Now replace atom dummy atom masses with the reparitioned mass
+            # from the opposite end state.
+
+            edit_mol = self._sire_object.edit()
```

Comment:
Should we be trying to use the new `sire` API in the new PRs? So e.g. something like:

```python
c = self._sire_object.cursor()
for atom in c.atoms():
    if "du" in atom["ambertype0"]:
        atom["mass0"] = atom["mass1"]
    elif "du" in atom["ambertype1"]:
        atom["mass1"] = atom["mass0"]
self._sire_object = Molecule(c.commit())
```
Here I've used a name check for the dummies but an element check might be more appropriate.

Reply:
It could, but in the existing Sire code (not the PR that is yet to be merged) you can't use a property map to index in this way, so the property naming _could_ be wrong. That said, I agree that it would be good (where appropriate) to start using the new API in PRs so that the code gets slowly ported over. I'd prefer to do it in self-contained functions, or with new functionality, so that we don't end up with a weird mixed-API that might be confusing for developers to follow. (In this case, it would be easy to replace the whole thing.)